### ISSUE TITLE
NullPointerException: null - displayFixedParticleEffect

### ISSUE THREAD

**Issue by mibby** [Esophose/PlayerParticles#11]
Compiled as of commit https://github.com/Esophose/PlayerParticles/commit/3d2306e7bb34b161d2ae281400e6d2b900078685

Using the 1.12 compiled version on Paper dev 1562 (Spigot 1.12.2). 

Edit: This also occurs with the latest spigot uploaded version.

@Esophose This error spams repeatedly at start-up, non-stop.

```
[06:26:59] [Server thread/WARN]: [PlayerParticles] Task #217 for PlayerParticles v5.2 generated an exception
java.lang.NullPointerException: null
	at com.esophose.playerparticles.manager.ParticleManager.displayFixedParticleEffect(ParticleManager.java:265) ~[?:?]
	at com.esophose.playerparticles.manager.ParticleManager.run(ParticleManager.java:209) ~[?:?]
	at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:63) ~[patched_1.12.2.jar:git-Paper-1562]
	at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:423) ~[patched_1.12.2.jar:git-Paper-1562]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:840) ~[patched_1.12.2.jar:git-Paper-1562]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:423) ~[patched_1.12.2.jar:git-Paper-1562]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:774) ~[patched_1.12.2.jar:git-Paper-1562]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:666) ~[patched_1.12.2.jar:git-Paper-1562]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
```

**Comment by mibby** [comment]
I purged the database sql tables and it seemed to have stopped the erroring. I guess a fixed 1.12 particle I had set is no longer functional with the same version.

**Comment by Esophose** [comment]
Because of all the material name changes and effect name changes, I decided to just delete all the data from the playerData.yml file and the database. The data didn't purge properly because you didn't have database-enable set to true when you loaded PlayerParticles v5.2 for the first time, as the data purge only happens when converting from a previous version to v5.2. You appear to have had a fixed effect stored in the database with an effect name that was changed between v5.1 and v5.2, which is what's causing the error. I'll add in a check in v5.3 that just deletes the fixed effect if it's found with an invalid effect, style, or data.

**Comment by mibby** [comment]
Alright, thanks. However it was PlayerParticles that reset the config back to default, disabling the first startup with database enabled. :( So I wouldn't have been able to keep the database enabled after updating to have it migrate. 

Deleting any bad fixed particles or data to prevent erroring is probably the best solution.

**Comment by Esophose** [comment]
Do you know of any way to transfer all the old configuration over the the new one without deleting all the comments? I'm able to transfer the settings over but then all of my comments get deleted from the config.yml file... That's why I end up resetting the entire config each update instead.

**Comment by mibby** [comment]
Trying to think what plugins I use that have auto config updaters.. I think FastAsyncWorldEdit _might_ auto-update without resetting. Any plugins that generally utilize a version number in their config also update and migrate settings without resetting.

There's probably a better way to do it, but this utility might help for now.
https://www.spigotmc.org/threads/configuration-updater.284804/

**Comment by Esophose** [comment]
Planned fix in v5.3, the entire database backend is getting rewritten and will require all the old data to be deleted again.
